lib/scopes/scopes.go

```diff
@@ -23,14 +23,15 @@ import (
 	"iter"
 	"regexp"
 	"strings"
+	"unicode"
 
 	"github.com/gravitational/trace"
 )
 
 // segmentRegexp is the regular expression used to validate scope segments. It allows
-// alphanumeric characters, hyphens, underscores, and periods. It also requires that the
-// segment starts and ends with an alphanumeric character.
-var segmentRegexp = regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9\-\_\.]*[a-zA-Z0-9]$`)
+// lowercase alphanumeric characters, hyphens, underscores, and periods. It also requires
+// that the segment starts and ends with an alphanumeric character.
+var segmentRegexp = regexp.MustCompile(`^[a-z0-9][a-z0-9\-\_\.]*[a-z0-9]$`)
```

Comment:
Is there a reason we don't allow segments to be a single character?

Reply:
The original reason is that I used a hostname segment regex as the starting point for this.  But given the intended UX around scopes and how the web UI will display them, I do think single-letter segments will end up being gross/confusing.